src/browser-polyfill.js

```diff
@@ -153,6 +150,27 @@ if (typeof browser === "undefined") {
               // Catch API parameters validation errors and rejects the promise.
               reject(error);
             }
+          }
+
+          if (metadata.fallbackToNoCallback) {
+            // This API method has currently no callback on Chrome, but it return a promise on Firefox,
+            // and so the polyfill try to pass a callback but fallback to not passing the callback
+            // if the first call have been rejected.
+            try {
+              target[name](...args, makeCallback({resolve, reject}, metadata));
+            } catch (cbError) {
+              console.warn(`${name} API method doesn't seem to support the callback parameter, ` +
+                           "falling back to call it without a callback: ", cbError);
+
+              // Update the API method metadata, so that the next API call will not try to
```

Comment:
Move this after line 170, to make sure that the API call is actually valid before considering the API callback-less.

Reply:
:+1: great call!